Binary files SDL2-widgets-2.1/bouncy-tune/bouncy-tune and originals/SDL2-widgets-2.1/bouncy-tune/bouncy-tune differ
diff -rupN SDL2-widgets-2.1/bouncy-tune/Makefile originals/SDL2-widgets-2.1/bouncy-tune/Makefile
--- SDL2-widgets-2.1/bouncy-tune/Makefile	2018-04-26 10:04:17.406849630 +0100
+++ originals/SDL2-widgets-2.1/bouncy-tune/Makefile	2014-05-30 09:51:05.000000000 +0100
@@ -1,7 +1,8 @@
-CC := i686-pc-btos-gcc
-CFLAGS := -g -O -std=gnu++11 -Wall -Wuninitialized -Wshadow\
-  -Wno-non-virtual-dtor -Wno-delete-non-virtual-dtor -Wno-multichar -I../../install/btos/include/SDL2
-LFLAGS := -lm -lstdc++ -lfreetype -L../../install/btos/lib -lSDL2 -lSDL2_ttf
+CC := g++
+# CFLAGS := -g $$(sdl2-config --cflags) -I.. -std=c++11 -O0 -Wno-multichar -fmax-errors=3
+CFLAGS := -g $$(sdl2-config --cflags) -I.. -std=c++11 -O \
+  -Wuninitialized -Wshadow -Wall -Wno-non-virtual-dtor -Wno-delete-non-virtual-dtor -Wno-multichar
+LFLAGS := $$(sdl2-config --libs) -lSDL2_ttf
 OBJ := bouncy-tune.o ../make-waves/dump-wave.o ../sdl-widgets.a
 
 .SUFFIXES=
diff -rupN SDL2-widgets-2.1/config.h originals/SDL2-widgets-2.1/config.h
--- SDL2-widgets-2.1/config.h	2018-04-26 10:04:17.406849630 +0100
+++ originals/SDL2-widgets-2.1/config.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-const char* FONTPATH="DroidSans.ttf";
-const char* FONTPATH_MONO="DroidSansMono.ttf";
Binary files SDL2-widgets-2.1/hello/hello and originals/SDL2-widgets-2.1/hello/hello differ
diff -rupN SDL2-widgets-2.1/hello/Makefile originals/SDL2-widgets-2.1/hello/Makefile
--- SDL2-widgets-2.1/hello/Makefile	2018-04-26 10:04:17.406849630 +0100
+++ originals/SDL2-widgets-2.1/hello/Makefile	2014-05-30 09:51:05.000000000 +0100
@@ -1,8 +1,6 @@
-CC := i686-pc-btos-gcc
-CFLAGS := -g -O -std=gnu++11 -Wall -Wuninitialized -Wshadow\
-  -Wno-non-virtual-dtor -Wno-delete-non-virtual-dtor -Wno-multichar -I../../install/btos/include/SDL2
-LFLAGS := -lm -lstdc++ -lfreetype -L../../install/btos/lib -lSDL2 -lSDL2_ttf
-OBJ := shapes.o sdl-widgets.o testsw.o
+CC := g++
+CFLAGS := -g $(shell sdl2-config --cflags) -std=c++11
+LFLAGS := $(shell sdl2-config --libs) -lSDL2_ttf
 
 .SUFFIXES=
 
diff -rupN SDL2-widgets-2.1/Makefile originals/SDL2-widgets-2.1/Makefile
--- SDL2-widgets-2.1/Makefile	2018-04-26 10:04:17.406849630 +0100
+++ originals/SDL2-widgets-2.1/Makefile	2014-06-28 14:13:04.000000000 +0100
@@ -1,8 +1,8 @@
-CC := i686-pc-btos-gcc
+CC := g++
 # CFLAGS :=-g -O0 $$(sdl2-config --cflags) -std=c++11 -Wno-multichar -fmax-errors=5
-CFLAGS := -g -O -std=gnu++11 -Wall -Wuninitialized -Wshadow\
-  -Wno-non-virtual-dtor -Wno-delete-non-virtual-dtor -Wno-multichar -I../install/btos/include/SDL2
-LFLAGS := -lm -lstdc++ -lfreetype -L../install/btos/lib -lSDL2 -lSDL2_ttf
+CFLAGS := -g -O $$(sdl2-config --cflags) -std=c++11 -Wall -Wuninitialized -Wshadow \
+  -Wno-non-virtual-dtor -Wno-delete-non-virtual-dtor -Wno-multichar
+LFLAGS := $$(sdl2-config --libs) -lSDL2_ttf
 OBJ := shapes.o sdl-widgets.o testsw.o
 
 .SUFFIXES=
diff -rupN SDL2-widgets-2.1/make-waves/Makefile originals/SDL2-widgets-2.1/make-waves/Makefile
--- SDL2-widgets-2.1/make-waves/Makefile	2018-04-26 10:04:17.406849630 +0100
+++ originals/SDL2-widgets-2.1/make-waves/Makefile	2014-09-09 08:18:22.000000000 +0100
@@ -1,7 +1,8 @@
-CC := i686-pc-btos-gcc
-CFLAGS := -g -O -std=gnu++11 -Wall -Wuninitialized -Wshadow\
-  -Wno-non-virtual-dtor -Wno-delete-non-virtual-dtor -Wno-multichar -I../../install/btos/include/SDL2
-LFLAGS := -lm -lstdc++ -lfreetype -L../../install/btos/lib -lSDL2 -lSDL2_ttf
+CC := g++
+CFLAGS := -g -O $$(sdl2-config --cflags) -I.. -std=c++11 \
+  -Wuninitialized -Wshadow -Wall -Wno-multichar -Wno-delete-non-virtual-dtor \
+  -Wno-delete-non-virtual-dtor -Wno-sequence-point
+LDFLAGS := $$(sdl2-config --libs) -lSDL2_ttf
 OBJ := make-waves.o dump-wave.o ../sdl-widgets.a
 # sdl2-config --cflags = -I/usr/local/include/SDL2 -D_REENTRANT
 # sdl2-config --libs   = -L/usr/local/lib -Wl,-rpath,/usr/local/lib -lSDL2 -lpthread
@@ -10,7 +11,7 @@ OBJ := make-waves.o dump-wave.o ../sdl-w
 
 make-waves: $(OBJ)
 	make -C .. archive
-	$(CC) $(OBJ) -o $@ $(LFLAGS)
+	$(CC) $(OBJ) -o $@ $(LDFLAGS)
 
 %.o: %.cpp
 	$(CC) -c $< $(CFLAGS)
Binary files SDL2-widgets-2.1/make-waves/make-waves and originals/SDL2-widgets-2.1/make-waves/make-waves differ
diff -rupN SDL2-widgets-2.1/sdl-widgets.cpp originals/SDL2-widgets-2.1/sdl-widgets.cpp
--- SDL2-widgets-2.1/sdl-widgets.cpp	2018-05-24 14:10:17.427279375 +0100
+++ originals/SDL2-widgets-2.1/sdl-widgets.cpp	2014-08-19 09:24:41.000000000 +0100
@@ -800,21 +800,20 @@ TopWin::TopWin(const char* wm_title,Rect
     texture(0),
     renderer(0),
     display_cmd(disp_cmd) {
-  int initresult = SDL_Init(init_flag|/*SDL_INIT_TIMER|*/SDL_INIT_VIDEO);
-  if (initresult < 0) err("SDL init problem: %s", SDL_GetError());
+  if (SDL_Init(init_flag|SDL_INIT_TIMER|SDL_INIT_VIDEO) < 0) err("SDL init problem");
 
   window=SDL_CreateWindow(wm_title,100,100,rect.w,rect.h,video_flag|SDL_WINDOW_SHOWN);
   if (accel) {
-    renderer=SDL_CreateRenderer(window,-1,SDL_RENDERER_SOFTWARE);
-    //if (!(texture=SDL_CreateTexture(renderer,SDL_GetWindowPixelFormat(window),SDL_TEXTUREACCESS_STREAMING,rect.w,rect.h)))
-    //  err("topw->texture: %s",SDL_GetError());
+    renderer=SDL_CreateRenderer(window,-1,SDL_RENDERER_ACCELERATED);
+    if (!(texture=SDL_CreateTexture(renderer,SDL_GetWindowPixelFormat(window),SDL_TEXTUREACCESS_STREAMING,rect.w,rect.h)))
+      err("topw->texture: %s",SDL_GetError());
   }
   if (!(render.surf=SDL_GetWindowSurface(window)))
     err("topw->render.surf: %s",SDL_GetError());
   if (!(render.render=SDL_CreateSoftwareRenderer(render.surf)))
     err("topw->render.render: %s",SDL_GetError());
-  //if (accel)
-  //  SDL_LockTexture(texture, 0, &render.surf->pixels, &render.surf->pitch);
+  if (accel)
+    SDL_LockTexture(texture, 0, &render.surf->pixels, &render.surf->pitch);
 
   if (TTF_Init() < 0) err("SDL ttf init problem");
   if (!(title_font=TTF_OpenFont(FONTPATH,nominal_font_size+1)))  // fontpath from config.h
Binary files SDL2-widgets-2.1/testsw and originals/SDL2-widgets-2.1/testsw differ
diff -rupN SDL2-widgets-2.1/testsw.cpp originals/SDL2-widgets-2.1/testsw.cpp
--- SDL2-widgets-2.1/testsw.cpp	2018-05-24 14:02:58.003287259 +0100
+++ originals/SDL2-widgets-2.1/testsw.cpp	2014-08-14 11:06:49.000000000 +0100
@@ -15,8 +15,7 @@
     You should have received a copy of the GNU Lesser General Public
     License along with this program.
 */
-#include <cmath>
-#include <cstdio>
+#include <math.h>
 #include "sdl-widgets.h"
 
 int col_R,col_G,col_B;
